File: backend/src/main/java/com/example/cinequest/entity/UserDetails.java
package com.example.cinequest.entity;

import jakarta.persistence.*;

import lombok.*;
import lombok.experimental.FieldDefaults;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
@Entity(name = "user_detail")
public class UserDetails {
    @Id
    @Column(name = "id")
    private String id;

    @Column(name = "username")
    private String username;

    @Column(name = "name")
    private String name;

    @Column(name = "surname")
    private String surname;

    @Column(name = "bio", length = 200)
    private String bio;

    @Column(name = "profile_photo")
    private String profilePhoto;
}

File: frontend/lib/src/domain/auth/entities/user_details.dart
import 'package:equatable/equatable.dart';

class UserDetails extends Equatable {
  const UserDetails({
    required this.id,
    this.username,
    this.name,
    this.surname,
    this.bio,
    this.profilePhoto,
  });
  final String id;
  final String? username;
  final String? name;
  final String? surname;
  final String? bio;
  final String? profilePhoto;

  @override
  List<Object?> get props => [
        id,
        username,
        name,
        surname,
        bio,
        profilePhoto,
      ];
}
